src/test/java/seedu/address/logic/commands/InternshipSortCommandTest.java

```diff
@@ -0,0 +1,128 @@
+
+        // same object -> returns true
+        assertTrue(sortCompanyNameCommand.equals(sortCompanyNameCommand));
+
+        // same values -> returns true
+        InternshipSortCommand sortNameCommandCopy = new InternshipSortCommand(fieldCompany, ascending);
+        assertTrue(sortCompanyNameCommand.equals(sortNameCommandCopy));
+
+        // different types -> returns false
+        assertFalse(sortCompanyNameCommand.equals(1));
+
+        // null -> returns false
+        assertFalse(sortCompanyNameCommand.equals(null));
+
+        // different field -> returns false
+        assertFalse(sortCompanyNameCommand.equals(sortLocationCommand));
+    }
+
+    @Test
+    public void execute_sortCompanyName_success() {
+        InternshipModel expectedModel = new InternshipModelManager();
+        Internship internship1 = internshipBuilder.withCompanyName("AAA").build();
+        model.addInternship(internship1);
+        Comparator<Internship> comparator = Internship.getComparator(fieldCompany, true);
+        model.sortFilteredInternshipList(comparator);
+        expectedModel.sortFilteredInternshipList(comparator);
+        assertEquals("AAA", model.getFilteredInternshipList().get(0).getCompanyName().toString());
+    }
+
+    @Test
+    public void execute_sortContactNumber_success() {
+        InternshipModel expectedModel = new InternshipModelManager();
+        Internship internship1 = internshipBuilder.withContactNumber("11111111").build();
+        model.addInternship(internship1);
+        Comparator<Internship> comparator = Internship.getComparator(fieldContactNumber, true);
+        model.sortFilteredInternshipList(comparator);
+        expectedModel.sortFilteredInternshipList(comparator);
+        assertEquals("11111111", model.getFilteredInternshipList().get(0).getContactNumber().toString());
+    }
+
+    @Test
+    public void execute_sortStatus_success() {
+        InternshipModel expectedModel = new InternshipModelManager();
+        Internship internship1 = internshipBuilder.withApplicationStatus("to_apply").build();
+        model.addInternship(internship1);
+        Comparator<Internship> comparator = Internship.getComparator(
+                InternshipSortCommandParser.FieldEnum.STATUS, true);
+        model.sortFilteredInternshipList(comparator);
+        expectedModel.sortFilteredInternshipList(comparator);
+        assertEquals("TO_APPLY", model.getFilteredInternshipList().get(0).getApplicationStatus().toString());
+    }
+
+    @Test
+    public void toString_validFieldAndOrder_success() {
+        InternshipSortCommand sortNameCommand = new InternshipSortCommand(fieldCompany, ascending);
+        String expected = InternshipSortCommand.class.getCanonicalName() + "{field="
+                + fieldCompany.toString() + ", order=" + ascending.toString() + "}";
+        assertEquals(expected, sortNameCommand.toString());
+    }
+}
```

Comment:
add sort by description, role, email, phone, remark
if possible add test to sort such that there's more than one entry, where the sorted list actually has a different order from filtered list (to check that sort actually changes the order properly)